test/codecs.test.js

```diff
@@ -236,7 +255,144 @@ QUnit.test('returns audio profile for default in audio group', function(assert)
       },
       'au1'
     ),
-    '5',
+    {audio: {type: 'mp4a', details: '.40.5'}},
     'returned parsed codec audio profile'
   );
 });
+
+QUnit.module('isVideoCodec');
+QUnit.test('works as expected', function(assert) {
+  [
+    'av1',
+    'avc01',
+    'avc1',
+    'avc02',
+    'avc2',
+    'vp09',
+    'vp9',
+    'vp8',
+    'vp08',
+    'hvc1',
+    'hev1',
+    'theora',
+    'mp4v'
+  ].forEach(function(codec) {
+    assert.ok(isVideoCodec(codec), `"${codec}" is seen as a video codec`);
+    assert.ok(isVideoCodec(` ${codec} `), `" ${codec} " is seen as video codec`);
+    assert.ok(isVideoCodec(codec.toUpperCase()), `"${codec.toUpperCase()}" is seen as video codec`);
+  });
+
+  ['invalid', 'foo', 'mp4a', 'opus', 'vorbis'].forEach(function(codec) {
+    assert.notOk(isVideoCodec(codec), `${codec} is not a video codec`);
+  });
+
+});
+
+QUnit.module('isAudioCodec');
+QUnit.test('works as expected', function(assert) {
+  [
+    'mp4a',
+    'flac',
+    'vorbis',
+    'opus',
+    'ac-3',
+    'ac-4',
+    'ec-3',
+    'alac'
+  ].forEach(function(codec) {
+    assert.ok(isAudioCodec(codec), `"${codec}" is seen as an audio codec`);
+    assert.ok(isAudioCodec(` ${codec} `), `" ${codec} " is seen as an audio codec`);
+    assert.ok(isAudioCodec(codec.toUpperCase()), `"${codec.toUpperCase()}" is seen as an audio codec`);
+  });
+
+  ['invalid', 'foo', 'bar', 'avc1', 'av1'].forEach(function(codec) {
+    assert.notOk(isAudioCodec(codec), `${codec} is not an audio codec`);
+  });
+});
+
+QUnit.module('muxerSupportCodec');
+QUnit.test('works as expected', function(assert) {
+  const validMuxerCodecs = [];
+  const invalidMuxerCodecs = [];
+
+  unsupportedMuxerCodecs.forEach(function(badCodec) {
+    invalidMuxerCodecs.push(badCodec);
+    supportedMuxerCodecs.forEach(function(goodCodec) {
+      invalidMuxerCodecs.push(`${goodCodec}, ${badCodec}`);
+    });
+  });
+
+  // generate all combinations of valid codecs
+  supportedMuxerCodecs.forEach(function(codec, i) {
+    validMuxerCodecs.push(codec);
+
+    supportedMuxerCodecs.forEach(function(subcodec, z) {
```

Comment:
Minor: but is subcodec a standard name for a different media type's codec?

Reply:
its not going to change this to `_codec` for clarity.